### PR TITLE
docs #980 schema registry API reference using redocusaurus p…

### LINKED ISSUE #980: Document Schema Registry API (and other APIs) within Redpanda docs
### Describe the Issue

The Schema Registry API docs currently link externally to SwaggerHub. For a consistent user experience, we should document APIs within Redpanda docs.

Other missing APIs: 
- HTTP proxy
- Admin API

Relates to #224  

### Updates to existing documentation

Document Schema Registry API (https://docs.redpanda.com/docs/platform/reference/schema-registry-api/) within Redpanda docs.

### PR DESCRIPTION
**Pull request by shorgi**
Fixes #980 

- Uses redocusaurus plugin to create schema registry API page from OpenAPI (JSON) spec.

Steps to generate docs from a OpenAPI/SwaggerUI spec:
1. Convert API spec into JSON
2. Save API spec in <redpanda-docs-repo>/static/api/
3. In <redpanda-docs-repo>/docusaurus.config.js in its presets for redocusaurus, add the API spec to specs, with its spec (path to API JSON file) and route (URL of generated doc).

Review deadline: Thu Jan 12

Preview: https://deploy-preview-981--redpanda-documentation.netlify.app/docs/platform/reference/api-reference/

### CODE REVIEW

**Comment by BenPope** [comment]
Looks great.

Can you describe the steps required to generate the docs from a specific version of the swagger spec?

It would be great to have that in the commit message, too.

**Comment by shorgi** [comment]
@BenPope amended commit message for https://github.com/redpanda-data/documentation/pull/981/commits/ffc2e5d65a3fcb273759ba5d839aad8789d424f6 with steps to generate doc from API spec

File: docs/platform/reference/api-reference.mdx [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+---
+
+<head>
+    <meta name="title" content="API References | Redpanda Docs"/>
+    <meta name="description" content="API References for Redpanda (schema registry)."/>
+</head>
+
+Use Redpanda's API reference documentation to learn about and interact with endpoints:
+
+- [Schema registry API reference](/docs/api/pandaproxy-schema-registry) (`pandaproxy-schema-registry`)
```
Comment (Feediver1) [inline docs/platform/reference/api-reference.mdx:12]:
```suggestion
- [Schema Registry API Reference](/docs/api/pandaproxy-schema-registry) (`pandaproxy-schema-registry`)
```

File: docs/platform/reference/api-reference.mdx [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+- [HTTP proxy API reference](/docs/api/pandaproxy-rest) (`pandaproxy-rest`)
```
Comment (Feediver1) [inline docs/platform/reference/api-reference.mdx:13]:
```suggestion
- [HTTP Proxy API Reference](/docs/api/pandaproxy-rest) (`pandaproxy-rest`)
```

File: static/api/pandaproxy-schema-registry.json [hunk 3]
```diff
@@ -0,0 +1,941 @@
...
+              "schema": {
+                "$ref": "#/definitions/error_body"
+              }
+            }
+          }
+        }
+      },
+      "/schemas/ids/{id}": {
+        "get": {
+          "summary": "Get a schema by id.",
```
Comment (Feediver1) [inline static/api/pandaproxy-schema-registry.json:239]:
```suggestion
          "summary": "Get a schema by ID.",
```

File: static/api/pandaproxy-schema-registry.json [hunk 4]
```diff
@@ -0,0 +1,941 @@
...
+              "schema": {
+                "$ref": "#/definitions/error_body"
+              }
+            }
+          }
+        }
+      },
+      "/schemas/ids/{id}/versions": {
+        "get": {
+          "summary": "Get a list of subject-version for the schema id.",
```
Comment (Feediver1) [inline static/api/pandaproxy-schema-registry.json:287]:
```suggestion
          "summary": "Get a list of subject-version for the schema ID.",
```

**Review by BenPope — COMMENTED** [review]
LGTM

File: static/api/pandaproxy-schema-registry.json [hunk 5]
```diff
@@ -0,0 +1,941 @@
...
+              "schema": {
+                "$ref": "#/definitions/error_body"
+              }
+            }
+          }
+        }
+      },
+      "/subjects/{subject}": {
+        "post": {
+          "summary": "Check if a schema is already registered for the subject.",
```
[the remark this answers is not in the document]
Comment (shorgi) [reply static/api/pandaproxy-schema-registry.json:374]: Will open PR.

Future improvement: create script to automatically concat the .json files from source and copy into docs.

File: static/api/pandaproxy-schema-registry.json [hunk 6]
```diff
@@ -0,0 +1,941 @@
...
+              "schema": {
+                "$ref": "#/definitions/error_body"
+              }
+            }
+          }
+        }
+      },
+      "/subjects/{subject}/versions/{version}/referencedBy": {
+        "get": {
+          "summary": "Retrieve a list of schema ids that reference the subject and version.",
```
Comment (Feediver1) [inline static/api/pandaproxy-schema-registry.json:747]:
```suggestion
          "summary": "Retrieve a list of schema IDs that reference the subject and version.",
```

**Review by Feediver1 — APPROVED** [review]
Good job @shorgi

**Pull request merged**